lib/kubernetes-deploy/rollout_conditions.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+module KubernetesDeploy
+  class RolloutConditionsError < StandardError
+  end
+
+  class RolloutConditions
+    class << self
+      def parse_conditions(conditions_string)
```

Comment:
Why did you decide to have parsing be a class method that returns a hash, instead of `from_conditions` method ?

Reply:
I think I agree with Danny's suggestion (`from_annotation` that returns `RolloutConditions` instance?), and I have a couple other ideas about this class I'd like to run by to you tomorrow if you have time. :)